### PR TITLE
replace assert with if

### PR DESCRIPTION
**Pull request by fabianegli**
- [x] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [x] I have read the ``CONTRIBUTING.rst`` file, have run ``pre-commit`` locally,
and understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [x] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

Relates to #3600

### CODE REVIEW

File: Bio/bgzf.py
Comment (peterjc) [inline Bio/bgzf.py:467]: The message is essentially for debugging, this would be more elegant:

```python
raise ValueError(f"{x_len=} and {extra_len=} differ")
```

Sadly that needs Python 3.8 onwards.

**Review by peterjc — APPROVED** [review]

File: Bio/bgzf.py
Comment (fabianegli) [reply Bio/bgzf.py:467]: I would have done that if 3.7 wasn't supported any more, but it still is if I am still up to date.

Comment (peterjc) [reply Bio/bgzf.py:467]: No, I was only waiting for the CI to finish before merging your contribution.

**Comment by peterjc** [comment]
Thank you

**Pull request merged**
